blocks/product-list-page/product-list-page.js

```diff
@@ -1,73 +1,90 @@
-      listview: true,
-      displayMode: '', // "" for plp || "PAGE" for category/catalog
-      addToCart: async (...args) => {
-        const { addProductsToCart } = await import('../../scripts/__dropins__/storefront-cart/api.js');
-        await addProductsToCart([{
-          sku: args[0],
-          options: args[1],
-          quantity: args[2],
-        }]);
-      },
+  // PLP Config
+  const plpConfig = {
+    pageSize: 8,
+    perPageConfig: {
+      pageSizeOptions: '12, 24, 36',
+      defaultPageSizeOption: '12',
     },
-    context: {
-      customerGroup: await getConfigValue('commerce.headers.cs.Magento-Customer-Group'),
+    minQueryLength: '2',
+    currencySymbol: '$',
+    currencyRate: '1',
+    displayOutOfStock: true,
+    allowAllProducts: false,
+    imageCarousel: false,
+    optimizeImages: true,
+    imageBaseWidth: 200,
+    listview: true,
+    currentCategoryUrlPath: type !== 'search' ? urlpath : null,
+    displayMode: '',
+    addToCart: async (...args) => {
+      const { addProductsToCart } = await import('../storefront-cart/api.js');
+      await addProductsToCart([{
+        sku: args[0],
+        options: args[1],
+        quantity: args[2],
+      }]);
     },
-    route: ({ sku, urlKey }) => {
-      const a = new URL(window.location.origin);
-      a.pathname = `/products/${urlKey}/${sku}`;
-      return a.toString();
+    route: {
+      route: '/search',
+      query: 'q',
     },
   };
 
-  if (type !== 'search') {
-    storeDetails.config.categoryName = document.querySelector('.default-content-wrapper > h1')?.innerText;
-    storeDetails.config.currentCategoryId = category;
-    storeDetails.config.currentCategoryUrlPath = urlpath;
+  // Get Config Values
+  const environmentId = await getConfigValue('commerce.headers.cs.Magento-Environment-Id');
+  const apiKey = await getConfigValue('commerce.headers.cs.x-api-key');
+  const apiUrl = await getConfigValue('commerce-endpoint');
+  const websiteCode = await getConfigValue('commerce.headers.cs.Magento-Website-Code');
+  const storeCode = await getConfigValue('commerce.headers.cs.Magento-Store-Code');
+  const storeViewCode = await getConfigValue('commerce.headers.cs.Magento-Store-View-Code');
+  const customerGroup = await getConfigValue('commerce.headers.cs.Magento-Customer-Group');
+
+  // Store Config
+  const storeConfig = {
+    type: 'eds',
+    environmentId,
+    environmentType: (async () => {
+      const endpoint = apiUrl;
+      return (endpoint.includes('sandbox')) ? 'testing' : '';
+    })(),
+    apiKey,
+    apiUrl,
+    websiteCode,
+    storeCode,
+    storeViewCode,
+    customerGroup,
+    route: ({ sku, urlKey }) => `/products/${urlKey}/${sku}`,
+    defaultHeaders: {
+      'Magento-Environment-Id': environmentId,
+      'Magento-Website-Code': websiteCode,
+      'Magento-Store-Code': storeCode,
+      'Magento-Store-View-Code': storeViewCode,
+      'Content-Type': 'application/json',
+      'X-Api-Key': apiKey,
```

Comment:
Oh I realize this should use dynamic behavior. Also not sure about use of `defaultHeaders` vs just `headers` as variable name.

```suggestion
      'Content-Type': 'application/json',
      ...(await getHeaders('cs'))
```

Reply:
I know you already define each above, but the point of dynamic `getHeaders` is incase customer wants to add new headers to `cs` query, for whatever reason. So it needs to just read whatever is in config and apply it.

Reply:
So the issue here is that PLP shim is not leveraging @adobe/fetch-graphql, so we don't have those functions available.  This is why we are setting the headers as such.  I'm hoping we don't need to do the effort to implement that library in this MVP version. Obviously, we would in GA version.
For CCDM functionality, default headers are sent all the time, and search headers are ones that can be overwritten using a search event.  This is also logic that should probably be done in a better way for GA version.

Reply:
>so we don't have those functions available.

Forget `setFetchGraphqlHeaders` or `setEndpoint` - this is just taking some properties from a map and passing them to your interface. You definitely have `getHeaders` as that is from the boilerplate. And it returns a map. Spreading the map will just create the same exact list you have here.

Or are you saying that if we spread the map, it may try to add headers which are not valid?

I think the logic should be that you define headers in config, then pass them to the dropin. Passing to dropin can be either via `setFetchGraphqlHeaders` or as some argument to the container. Either is fine for MVP. What I _disagree_ with, is that you need to define this `searchHeaders` object. I do not see why the dropin should care what headers it attaches to it's GQL call. If headers are defined on the container/props, it should use them.

You are exposing internal workings of the application by having `searchHeaders` even be a thing. IMO that is an antipattern.

Reply:
>For CCDM functionality, default headers are sent all the time, and search headers are ones that can be overwritten using a search event. This is also logic that should probably be done in a better way for GA version.

I think I need to see how this is working internally on the dropin. Inside your dropin are you listening for this event? 

I was under impression that page would reload, and all that matters are the initial header values/props passed.